assets/js/amp-service-worker-offline-commenting.js

```diff
@@ -0,0 +1,45 @@
+/* global ERROR_MESSAGES, SITE_URL */
+{
+	const queue = new wp.serviceWorker.backgroundSync.Queue( 'amp-wpPendingComments' );
+	const errorMessages = ERROR_MESSAGES;
+
+	const commentHandler = ( { event } ) => {
+		const clonedRequest = event.request.clone();
+		return fetch( event.request ).catch( () => {
+			return clonedRequest.blob().then( ( body ) => {
+				const queuedRequest = new Request( event.request.url, {
+					method: event.request.method,
+					headers: event.request.headers,
+					mode: event.request.mode,
+					credentials: event.request.credentials,
+					referrer: event.request.referrer,
+					redirect: 'follow',
+					body: body
+				} );
+
+				// Add request to queue. @todo Replace when upgrading to Workbox v4!
```

Comment:
@miina Is this still valid?

Reply:
@westonruter I believe so, that was related to https://github.com/GoogleChrome/workbox/pull/1710 specifically. Looks like it got merged meanwhile.

Reply:
It's required sure enough. The offline commenting is currently failing:

![image](https://user-images.githubusercontent.com/134745/49178146-d8a5dc00-f303-11e8-92ef-7c29b4ba87ad.png)

Reply:
Fixed in https://github.com/Automattic/amp-wp/pull/1519/commits/1e77c6e56f3ad0b8a39a988f4aab409bb646e326 and https://github.com/xwp/pwa-wp/pull/109

Reply:
Correction: https://github.com/Automattic/amp-wp/pull/1519/commits/ee465e718f52f6ea90d102a9b033240aa24505a6